File: soap/client_soap.js
// SOAP
const soap = require('strong-soap').soap;

const url = 'http://localhost:8000/soap?wsdl';

const requestArgs = {
  a: 1,
  b: 7
};

const id_user = {
  id_usuario: 1
};
const id_play = {
  id_playlist: 1
};
const id_music = {
  id_musica: 1
};

soap.createClient(url, {}, (err, client) => {
  if (err) {
    return console.error('Erro ao criar cliente SOAP:', err);
  }

  client.Listar_Usuarios((err, result) => {
    if (err) {
      console.error('Erro na chamada SOAP:', err);
    } else {
      console.log('Resultado - Listar_Usuarios:\n', JSON.parse(result.result));
    }
  });

  client.Listar_Musicas((err, result) => {
    if (err) {
      console.error('Erro na chamada SOAP:', err);
    } else {
      console.log('Resultado - Listar_Musicas:\n', JSON.parse(result.result));
    }
  });

  client.Listar_Playlist_Usuario(id_user, (err, result) => {
    if (err) {
      console.error('Erro na chamada SOAP:', err);
    } else {
      console.log('Resultado - Listar_Playlist_Usuario:\n', JSON.parse(result.result));
    }
  });

  client.Listar_Musicas_Playlist(id_play, (err, result) => {
    if (err) {
      console.error('Erro na chamada SOAP:', err);
    } else {
      console.log('Resultado - Listar_Musicas_Playlist:\n', JSON.parse(result.result));
    }
  });

  client.Listar_Playlist_Musica(id_music, (err, result) => {
    if (err) {
      console.error('Erro na chamada SOAP:', err);
    } else {
      console.log('Resultado - Listar_Playlist_Musica:\n', JSON.parse(result.result));
    }
  });
});

File: soap/client_soap.py
from zeep import Client
import json

# URL do WSDL gerado pelo seu servidor Node.js
wsdl_url = "http://localhost:8000/soap?wsdl"

# Criação do cliente SOAP
client = Client(wsdl=wsdl_url)

# Chamada para Listar_Usuarios
print("Listar_Usuarios:")
usuarios = json.loads(client.service.Listar_Usuarios())
for item in usuarios:
  print(item)

# Chamada para Listar_Musicas
print("Listar_Musicas:", )
musicas = json.loads(client.service.Listar_Musicas())
for item in musicas:
  print(item)
  
# Chamada para Listar_Playlist_Usuario
print("Listar_Playlist_Usuario:", )
playlists = json.loads(client.service.Listar_Playlist_Usuario(id_usuario=1))
for item in playlists:
  print(item)
  
# Chamada para Listar_Musicas_Playlist
print("Listar_Musicas_Playlist:", )
musicas_play = json.loads(client.service.Listar_Musicas_Playlist(id_playlist=1))
for item in musicas_play:
  print(item)

# Chamada para Listar_Playlist_Musica
print("Listar_Playlist_Musica:", )
play_music = json.loads(client.service.Listar_Playlist_Musica(id_musica=1))
for item in play_music:
  print(item)

